### ISSUE TITLE
tfe_variable resources broken in 0.12

### ISSUE THREAD

**Issue by markercm** [terraform-providers/terraform-provider-tfe#126]
When I run a plan after 0.12 it appears to not be able to detect the existence/state of the tfe_variables that I have defined.

**Comment by markercm** [comment]
Solved the issue by locking to previous version.

  version  = "v0.11.4"

**Comment by lafentres** [comment]
Hi! Could you please provide the Terraform version you are on, your config (or an example where you can reproduce the problem), and a description of the things you did when you encountered this error?

**Comment by markercm** [comment]
We are using TFE, the workspace we use the tfe provider on is using the 0.12.2 binary.

`## This Workspace will migrate the test state from old local Terraform to Terraform Enterprise. 

resource "tfe_workspace" "terraform101" {
  name         = "terraform101-ws"
  organization = "myorg"
  vcs_repo {
    identifier     = "terraform101/terraform-ghe-reposforclass"
    branch         = "master"
    oauth_token_id = var.oauth_token_id
  }
}

resource "tfe_variable" "terraform101-ws-tenid" {
  key          = "ARM_TENANT_ID"
  value        = var.tenant_id
  category     = "env"
  workspace_id = "${tfe_workspace.terraform101.id}"
  sensitive    = false
}`

Ran a plan and it wanted to Create the same resources that were created earlier the same day.

Ran apply:
`Error: Error creating env variable ARM_TENANT_ID: resource not found

  on terraform101.tf line 13, in resource "tfe_variable" "terraform101-ws-tenid":
  13: resource "tfe_variable" "terraform101-ws-tenid" {
`

After encountering this, I set the above tfe provider block version then had to manually delete all of the variables that were previously defined so the create would succeed.

In another tfe repo, same config, I hadn't run a plan today, so set the version lock on the tfe provider and ran a plan and no changes were present.

**Comment by lafentres** [comment]
Thanks! One more question: what version of TFE are you on?

**Comment by markercm** [comment]
Version v201911-3

**Comment by philbal611** [comment]
We're also effected by this. We're currently seeing all 140 of our `tfe_variable` resources attempting to be created in the plan. I've been able to verify that the resources are still in tact and available in state, i.e.:

```hcl
+ module.workspace-example.tfe_variable.arm_client_id
      id:           <computed>
      category:     "env"
      hcl:          "false"
      key:          "ARM_CLIENT_ID"
      sensitive:    "false"
      workspace_id: "Org/example"
```
```hcl
terraform state show module.workspace-example.tfe_variable.arm_client_id
id           = var-vPbaAhxMxdELd3u9
category     = env
hcl          = false
key          = ARM_CLIENT_ID
sensitive    = false
value        = <client_id>
workspace_id = Org/example
```

Reverting to `0.11.4` has worked for us in the meantime as well. We're running TFE `v201912-3` & TF `0.11.14`

**Comment by josh-barker** [comment]
I've also been effected by the `0.12.0` release.

We're running TFE Version: `v201912-4`

It looks like the provider is using newer endpoints that aren't supported by TFE.

Both of these return errors
- /api/v2/workspaces/:workspace_id/vars
```shell
curl \
  --header "Authorization: Bearer $TOKEN" \
  --header "Content-Type: application/vnd.api+json" \
"https://terraform.somewhere.com/api/v2/workspaces/ws-XXXXXXXXXXXXXXXX/vars"
{"errors":[{"status":"404","title":"not found"}]}
```
- /api/v2/workspaces/:workspace_id/vars/:variable_id
```bash
curl \
  --header "Authorization: Bearer $TOKEN" \
  --header "Content-Type: application/vnd.api+json" \
"https://terraform.somewhere.com/api/v2/workspaces/ws-XXXXXXXXXXXXXXXX/vars/var-XXXXXXXXXXXXXXXX"
{"errors":[{"status":"404","title":"not found"}]}
```

Terraform Logs:

```
2020-02-12T11:35:26.827+1100 [DEBUG] plugin.terraform-provider-tfe_v0.12.0_x4: GET /api/v2/workspaces/ws-XXXXXXXXXXXXXXXX/vars/var-XXXXXXXXXXXXXXXX HTTP/1.1
...
...
2020-02-12T11:35:26.830+1100 [DEBUG] plugin.terraform-provider-tfe_v0.12.0_x4: 2020/02/12 11:35:26 [DEBUG] TFE API Response Details:
2020-02-12T11:35:26.830+1100 [DEBUG] plugin.terraform-provider-tfe_v0.12.0_x4: ---[ RESPONSE ]--------------------------------------
2020-02-12T11:35:26.830+1100 [DEBUG] plugin.terraform-provider-tfe_v0.12.0_x4: HTTP/2.0 404 Not Found
```

**Comment by caseylang** [comment]
Thank you all for all the helpful debugging information!
 
We've determined that this is due to incompatibility between tfe provider v0.12.0 and any version of TFE < 202001-1. We'll be updating the changelog to reflect this and releasing a patch update for the provider to warn users of the incompatibility.

Version 0.12.0 of the tfe provider included changes to support updates to the workspace variables API. These changes were introduced to Terraform Enterprise in v202001-1.

To resolve this, you can either upgrade to Terraform Enterprise v202001-1 or pin the tfe provider version to 0.11.x releases.

If you are using Terraform CLI version 0.11.x, the following block will pin the tfe provider to 0.11.x versions.
```
provider "tfe" {
  version = "~> 0.11.0"
  ...
}
```

If you are using Terraform CLI version 0.12.x, the following block will pin the tfe provider to 0.11.x versions.
```
terraform {
  required_providers {
    tfe = "~> 0.11.0"
  }
}
```

**Comment by caseylang** [comment]
Resolved with the release of [0.12.1](https://github.com/terraform-providers/terraform-provider-tfe/releases/tag/v0.12.1)
